src/Main.elm

```diff
@@ -124,17 +148,27 @@ update msg model =
                 selected =
                     case selection of
                         Just movie ->
-                            Selected movie
+                            Movie.Selected movie
 
                         Nothing ->
-                            NothingToSelect
+                            Movie.NothingToSelect
             in
-                ( { model
-                    | unwatched = remaining
-                    , selectedMovie = selected
-                  }
-                , Cmd.none
-                )
+                updateModal { model | unwatched = remaining } selected
```

Comment:
Minor, but maybe this should be `openModal`, since it's only used to open the modal, not to update it after data is loaded.  That also implies it only needs to take `Movie` as input rather than `MovieSelection`, since the other path isn't used (which I think is fine).  Would also eliminate the need for the `case` expression on 122

Reply:
I made it `updateModal` since it is used by the close button on the modal with a `NotSelected` to close itself.

Reply:
Admittedly, that's in a roundabout way. I actually pass `FocusMovie` to the modal so that it can send that message to close, which in turn calls updateModal.